### ISSUE TITLE
Login to ghcr.io no longer work in rc2 using a dockerconfig

### ISSUE THREAD

**Issue by souleb** [open-component-model/ocm#400]
**What happened**:

Using ocm `rc1` I was able to connect to ghcr.io to handle my components. That is no longer the case with rc2.  I get the following error

```shell
failed to list versions of component "ocm.software/mpas/bootstrap": failed to authorize: failed to fetch oauth token: unexpected status from POST request to https://ghcr.io/token: 405 Method Not Allowed
```

**What you expected to happen**:

I was expecting the same behaviour than `rc1` and to be able to list my component version.

**How to reproduce it (as minimally and precisely as possible)**:

on branch `bootstrap install` in `mpas` repository runt he following using a fresh `kind` cluster (it use the kubeconfig to bootstrap).

```sh
⋊> ~/g/s/g/o/MPAS on bootstrap-install ⨯ make build-dev  
⋊> ~/g/s/g/o/MPAS on bootstrap-install ⨯ ./bin/mpas bootstrap github --owner xxx --repository=xxx --personal --registry ghcr.io/souleb/mpas-bootstrap-component --target clusters/my-cluster 
````

It should just work.

then updgrade ocm to `rc2` to see the error

**Anything else we need to know**:

I create my repository as follows

```go
func makeOCIRepository(octx ocm.Context, repositoryURL, dockerconfigPath string) (ocm.Repository, error) {
	if !strings.Contains(repositoryURL, "https://") && !strings.Contains(repositoryURL, "http://") {
		repositoryURL = "https://" + repositoryURL
	}
	regURL, err := url.Parse(repositoryURL)
	if err != nil {
		return nil, err
	}

	spec := dockerconfig.NewRepositorySpec(dockerconfigPath, true)
	// attach the repository to the context, this propagates the consumer ids.
	_, err = octx.CredentialsContext().RepositoryForSpec(spec)
	if err != nil {
		return nil, fmt.Errorf("cannot access docker config: %w", err)
	}

	meta := ocireg.NewComponentRepositoryMeta(strings.TrimPrefix(regURL.Path, "/"), ocireg.OCIRegistryURLPathMapping)
	targetSpec := ocireg.NewRepositorySpec(regURL.Host, meta)
	target, err := octx.RepositoryForSpec(targetSpec)
	if err != nil {
		return nil, err
	}
	return target, nil
}
```

**Environment**:

**Comment by shivenduverma-sap** [comment]
Hi @souleb 
FYI, The latest version of ocm https://github.com/open-component-model/ocm/blob/main/pkg/contexts/ocm/repositories/ocireg/type.go#L37

> 	meta := ocireg.NewComponentRepositoryMeta(strings.TrimPrefix(regURL.Path, "/"), ocireg.OCIRegistryURLPathMapping)
> 	targetSpec := ocireg.NewRepositorySpec(regURL.Host, meta)

has already been done with the NewRepositorySpec(regURL.Host) call

**Comment by jensh007** [comment]
Related to  #368?

**Comment by shivenduverma-sap** [comment]
@souleb Can you point me to the place where you are using this specific URL? "https://ghcr.io/token"

**Comment by souleb** [comment]
I don't use that URL. I don't actually know why `ocm` request a token as I provide one with the cmd line. You can try the command I provided to test it.

Well the functions signatures are the same, so I'm expecting same behaviour on `rc2`.

**Comment by shivenduverma-sap** [comment]
--registry ghcr.io/souleb/mpas-bootstrap-component
Can you try with another registry, eg from Google Cloud? or AWS ECR?

**Comment by souleb** [comment]
I tried with `--registry ghcr.io/open-component-model/mpas-bootstrap-component`

```shell
⋊> ~/g/s/g/o/MPAS on bootstrap-install ⨯ ./bin/mpas bootstrap github --owner souleb --repository=mpas-test --personal --registry ghcr.io/open-component-model/mp
as-bootstrap-component --path clusters/my-cluster
Running mpas bootstrap ...
 ✓   Preparing Management repository mpas-test with branch main and visibility public
 ✗   Fetching bootstrap component from ghcr.io/open-component-model/mpas-bootstrap-component 
error: failed to fetch bootstrap component references: failed to fetch latest version of component "ocm.software/mpas/bootstrap": failed to list versions of component "ocm.software/mpas/bootstrap": failed to authorize: failed to fetch oauth token: unexpected status from POST request to https://ghcr.io/token: 405 Method Not Allowed
⋊> ~/g/s/g/o/MPAS on bootstrap-install ⨯    
```

Those 2 are public packages, we shouldn't have this error at all. And I use `ghcr.io` because that's where we host the bootstrap component.

**Comment by mandelsoft** [comment]
I tried various combinations with docker config, ocm config and implicitly used GITHUB_TOKEN. All those variations worked as expected for the CLI on the main branch.

Could you verify, that the problem you mentioned occurs for the CLI version on the main branch, also?

There was a problem with the usage of GITHUB_TOKEN some months ago, but this should be fixed in the meantime.

**Comment by mandelsoft** [comment]
I just figured out, that there was a change i the credential handling influencing the precedence of the various sources,

Now the lib distinguishes between explicit dedicated settings and other providers providing whole groups of credentials.
The docker config in now such a group, but was mapped to explicit settings in an earlier version.

The implicit handling of the env var GITHUB_TOKEN is still an explicit setting.

Therefore the docker config will not overwrite the implicit anymore, but both coexist.
The precedence during the evaluation is:
- first the explicit settings
- then the group settings.

Therefore setting the GITHUB_TOKEN will always have precedence over the docker config settings.

If you but explicit settings into the .ocmconfig, then these settings are explicit settings and therefore overwrite the implicit setting (here GITHUB_TOKEN.

So, if you have set GITHUB_TOKEN to a wrong value, but have a correct one in your dockerconfig, the the token will win.
The only way to correct this, is to put the correct creds into the .ocmconfig, or to explicitly call
the `SetCredentialsForConsumer` on your credentials context, or to apply an appropriate credentials config object to your config context.

I don't know how your concrete settings look like in detail, therefore I explained the process. So, what has change is
the precedence of GITHUB_TOKE over the docker config, may be this caused your trouble.
